### PR TITLE
Remove ImageMagick bug workaround

### PR DESCRIPTION
**Pull request by PhilipVinc**
Thanks to [ImageMagick#180](https://github.com/JuliaIO/ImageMagick.jl/pull/180) I can remove the ugly hack I had in place around for loading pngs from an in-memory stream.

This is fixed on ImageMagick v1.1.4. Which is an indirect dependency loaded by FileIO when needed on Linux (not on MacOs).  
However, I am unsure on how to properly set dependency bounds in this case.

Maybe we should carry the hack around for a bit longer?

cc @Arkoniak and @timholy who might have an idea about how to do this properly.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
It seems that ImageIO requires Julia >= 1.3, and FileIO requires ImageIO to load PNG all the time for some reason unknown to me, but it is not available in Julia 1.0

**Comment by timholy** [comment]
If you need a particular version of ImageMagick you can put it in your `[compat]` section even if the package itself doesn't directly `using ImageMagick`. (You have to list it in `[deps]` too.)

ImageIO depends on Julia 1.3+ for a good reason (the new artifact system), so you'll have to decide how you want to go forward: either abandon further updates for earlier Julia versions or don't explicitly require ImageIO.

FileIO should not require ImageIO if there is an alternative like ImageMagick. If you see otherwise please submit the full error message.

**Comment by PhilipVinc** [comment]
Thanks @timholy 

Regarding the requirement of ImageIO, please have a look at this failing Travis Job on Julia 1.0, linux
https://travis-ci.org/github/PhilipVinc/TensorBoardLogger.jl/jobs/675194294

 Relevant Installed packages are
```
 Installed ImageMagick ────────── v0.7.5
393
 Installed FileIO ───────────── v1.2.4
316
```

FileIO is called by the following code:
```julia
function deserialize_image_summary(summary)
     img = summary.image 
     value = load(Stream(format"PNG", IOBuffer(img.encoded_image_string)))     
     return value
end
```
and the error is 
```
All errors:
517===========================================
518ArgumentError: Package ImageIO not found in current path:
519- Run `import Pkg; Pkg.add("ImageIO")` to install the ImageIO package.
520
521===========================================
522MethodError: no method matching fd(::Base.GenericIOBuffer{Array{UInt8,1}})
523Closest candidates are:
524  fd(!Matched::GZip.GZipStream) at /home/travis/.julia/packages/GZip/JNmGn/src/GZip.jl:304
525  fd(!Matched::Base.Filesystem.File) at filesystem.jl:237
526  fd(!Matched::IOStream) at iostream.jl:36
527===========================================
528
529Fatal error:
530Deserialization_simple: Error During Test at /home/travis/build/PhilipVinc/TensorBoardLogger.jl/test/deserialization.jl:9
531  Got exception outside of a @test
532  ArgumentError: Package ImageIO not found in current path:
533  - Run `import Pkg; Pkg.add("ImageIO")` to install the ImageIO package.
534  
535  Stacktrace:
536   [1] handle_error(::ArgumentError, ::Stream{DataFormat{:PNG},Base.GenericIOBuffer{Array{UInt8,1}}}) at /home/travis/.julia/packages/FileIO/Glbhe/src/error_handling.jl:82
537   [2] handle_exceptions(::Array{Any,1}, ::String) at /home/travis/.julia/packages/FileIO/Glbhe/src/error_handling.jl:77
538   [3] #load#27(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::FileIO.Formatted) at /home/travis/.julia/packages/FileIO/Glbhe/src/loadsave.jl:189
539   [4] load at /home/travis/.julia/packages/FileIO/Glbhe/src/loadsave.jl:169 [inlined]
540   [5] deserialize_image_summary at /home/travis/build/PhilipVinc/TensorBoardLogger.jl/src/Deserialization/images.jl:4 [inlined]
541

**Comment by timholy** [comment]
FileIO will keep trying until something succeeds, but if nothing succeeds then it will complain about missing the first package.

It's only the fact that the `fd` also throws a MethodError. Fix that and the absence of ImageIO shouldn't  be a problem.

**Comment by timholy** [comment]
You need v1.1.4 of ImageMagick. You've got 0.7.5.

**Comment by timholy** [comment]
You could backport that fix to an earlier release cycle (but the onus is on you to do it).

**Comment by PhilipVinc** [comment]
Pardon.
I had not realised travis was installing such an old version of imagemagick.
I don't understand why. There should be no dependencies blocking a more recent version.

**Comment by timholy** [comment]
Current ImageMagick is Julia 1.3+

File: Project.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 
 [compat]
 ColorTypes = "0.7, 0.8, 0.9, 0.10"
-FileIO = "1"
+FileIO = ">= 1.2.3"
```
Comment (johnnychen94) [inline Project.toml]:
```suggestion
FileIO = "1.2.3"
```
which means `[1.2.3, 2.0.0)`

File: src/Deserialization/audio.jl [hunk 2]
```diff
@@ -2,7 +2,7 @@ function deserialize_audio_summary(summary)
     audio = summary.audio
 
     if audio.content_type == "audio/wav"
-        value = load(Stream(format"WAV", PipeBuffer(audio.encoded_audio_string)))
+        value = load(Stream{format"WAV"}(PipeBuffer(audio.encoded_audio_string)))
```
Comment (johnnychen94) [inline src/Deserialization/audio.jl]: This format requires FileIO >1.6.0 😆

File: src/Deserialization/images.jl [hunk 3]
```diff
@@ -1,7 +1,7 @@
 function deserialize_image_summary(summary)
     img = summary.image
 
-    value = _deserialize_png(img.encoded_image_string)
+    value = load(Stream(format"PNG", IOBuffer(img.encoded_image_string)))
```
Comment (johnnychen94) [inline src/Deserialization/images.jl]: and this requires FileIO < 1.6.0 😆

File: src/Deserialization/audio.jl (hunk 2 above)
Comment (johnnychen94) [reply src/Deserialization/audio.jl]: My patch to this is

```julia
if isdefined(FileIO, :action)
    # FileIO >= 1.6
    _format_stream(format, io) = FileIO.Stream{format}(io)
else
    _format_stream(format, io) = FileIO.Stream(format, io)
end
```

Comment (PhilipVinc) [reply src/Deserialization/audio.jl]: thanks!

Comment (PhilipVinc) [reply src/Deserialization/audio.jl]: would it make sense to support only FileIO>=1.6 ? Or that is too recent?

Comment (johnnychen94) [reply src/Deserialization/audio.jl]: Your call, I personally prefer to add a compat patch to avoid unnecessary version resolving locks.

**Pull request merged**